### PR TITLE
Issue #13095: kill mutation for VariableDeclarationUsageDistanceCheck 3

### PR DESCRIPTION
**Pull request by Kevin222004**
**1**  Issue #13095: kill mutation for VariableDeclarationUsageDistanceCheck 3

 **2** Link to Check Documentation :-
 https://checkstyle.org/config_coding.html#VariableDeclarationUsageDistance

**3** covering 
https://github.com/checkstyle/checkstyle/blob/6f451fbb49afc21b0e04b46c0576c7a984419a88/config/pitest-suppressions/pitest-coding-1-suppressions.xml#L129-L136

**4** Logic behind mutation removal
I don't find any test case which can kill this mutation.

https://github.com/Kevin222004/checkstyle/blob/d60bae5293a4221f4e5b28da1300441a90be5d44/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java#L569-L571
whenever the token type is off `TokenTypes.VARIABLE_DEF` then `resultDist++;` will increment by 1.

In default case 
https://github.com/Kevin222004/checkstyle/blob/d60bae5293a4221f4e5b28da1300441a90be5d44/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java#L589-L595
 `if (examineNode.findFirstToken(TokenTypes.SLIST) == null) {`  here if the `if` statment is been executing then similarly `resultDist++;` will increment by 1 their is no case possible where `SLIST` become the child of `VARIABLE_DEF` so it is always going to be null and every time only if will execute.

**5**  Reports :- 
https://kevin222004.github.io/reports/M3/2023-05-31-T-09-24-56/test-report/index.html

### CODE REVIEW

**Comment by Kevin222004** [comment]
I don't find any test case which can kill this mutation.

https://github.com/Kevin222004/checkstyle/blob/d60bae5293a4221f4e5b28da1300441a90be5d44/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java#L569-L571
whenever the token type is off `TokenTypes.VARIABLE_DEF` then `resultDist++;` will increment by 1.

In default case 
https://github.com/Kevin222004/checkstyle/blob/d60bae5293a4221f4e5b28da1300441a90be5d44/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java#L589-L595
 `if (examineNode.findFirstToken(TokenTypes.SLIST) == null) {`  here if the `if` statment is been executing then similarly `resultDist++;` will increment by 1 their is no case possible where `SLIST` become the child of `VARIABLE_DEF` so it is always going to be null and every time if will execute.

**Comment by Kevin222004** [comment]
Regression :- https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/8bfe682_2023155308/reports/diff/index.html

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java:589]: can just remove such case? instead of attaching it to default ?

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java
Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java]: done, removed

**Review by romani — APPROVED** [review]
ok to merge.

I think we need to extend even more project set for cases that we can not find code to cover and ready to apply pitest suggestion.
It will be not CI execution, but we can ask @Kevin222004 , to add 20 more HUGE java projects and let hit execute on his local validation and double confirm that nothing is found.

@Kevin222004 , if you agree please do this while we waiting for other mentors to review this PR.

**Comment by romani** [comment]
@Kevin222004 , conflict again .
please review what we can do to avoid conflicts, we use to did extra spaces in files between blocks to let git not mark them as conflict.

**Comment by nrmancuso** [comment]
>  to add 20 more HUGE java projects and let hit execute on his local validation and double confirm that nothing is found

@Kevin222004 do not forget about option to fork repo at https://github.com/nrmancuso/nrmancuso.github.io to generate and host lots of reports easily.

**Comment by romani** [comment]
We probably need to invest more in tool like this, as we will need even during regular feature development, chasing a regression will be always required.

**Comment by Kevin222004** [comment]
Thanks @nrmancuso I have this in my mind and will do it.

**Comment by Kevin222004** [comment]
Regression :- https://kevin222004.github.io/Regression/

@romani please look at this report I have chosen some of the projects.

**Comment by romani** [comment]
I am ok.

**Comment by Vyom-Yadav** [comment]
@Kevin222004 It is better to create an extended list and keep it on a gist or something. You can add these projects too:
```
https://github.com/square/retrofit
https://github.com/NationalSecurityAgency/ghidra
https://github.com/apache/dubbo
https://github.com/PhilJay/MPAndroidChart
https://github.com/jeecgboot/jeecg-boot
https://github.com/skylot/jadx
https://github.com/airbnb/lottie-android
https://github.com/dbeaver/dbeaver
https://github.com/netty/netty
https://github.com/zxing/zxing
https://github.com/Blankj/AndroidUtilCode
```
Please attach a regression report with these projects too.

**Comment by Kevin222004** [comment]
@Vyom-Yadav done
https://kevin222004.github.io/reports/M3/2023-05-30-T-17-54-50/test-report/index.html

**Comment by Vyom-Yadav** [comment]
@Kevin222004 Let's follow some pattern for the PR description.

1. Resolves #XXX or simply issue number if linking to parent issue
2. Link to Check Documentation
3. Mutations Killed, permalink from GitHub
4. Logic behind mutation removal (Can be either of)
4.1 Whether test cases were extended
4.2 Whether the code wasn't required with rationale
4.3 Extra code was added to support the extending code, hence killing the mutation
5. All reports
6. Additional Comments

This would make it easier to evaluate PRs throughout the project and easier to refer to them in the future.

**Comment by Vyom-Yadav** [comment]
@Kevin222004 All the configurations in the diff report don't use `allowedDistance` property. By default, it is `3` which may hide some violations we weren't expecting.

Please follow the pattern above and generate reports with `allowedDistance` as `1`.

**Review by Vyom-Yadav — APPROVED** [review]
Logically it looks correct to me. `SLIST` cannot be the first level child of `VARIABLE_DEF`. Still, let's generate a report for a sanity check. 

LGTM if new diff reports are clean.

**Comment by romani** [comment]
@Vyom-Yadav , as you finish review reassign it to next reviewer, you should have permission to do this now

**Comment by Kevin222004** [comment]
> Logically it looks correct to me. `SLIST` cannot be the first level child of `VARIABLE_DEF`. Still, let's generate reports for a sanity check.
> 
> LGTM if new diff reports are clean.

Regression :- https://kevin222004.github.io/reports/M3/2023-05-31-T-09-24-56/test-report/index.html

**Review by rdiachenko — APPROVED** [review]
lgtm

**Pull request merged**
